server/plugin/api.go

```diff
@@ -359,9 +359,105 @@ func (p *Plugin) handleSubscriptionNotifications(w http.ResponseWriter, r *http.
+	case constants.WorkItemCreated, constants.WorkItemUpdated, constants.WorkItemDeleted, constants.WorkItemCommented:
+		attachment = &model.SlackAttachment{
+			Text: body.DetailedMessage.Markdown,
+		}
+	case constants.PullRequestCreated, constants.PullRequestUpdated, constants.PullRequestMerged:
+		reviewers := "None" // When no reviewers are added to the pull request
+		for i := 0; i < len(body.Resource.Reviewers); i++ {
+			if i == 0 {
+				reviewers = ""
+			}
+
+			if i != len(body.Resource.Reviewers)-1 {
+				reviewers += fmt.Sprintf("%s, ", body.Resource.Reviewers[i].DisplayName)
+			} else {
+				reviewers += body.Resource.Reviewers[i].DisplayName
+			}
+		}
+
+		attachment = &model.SlackAttachment{
+			Pretext: body.Message.Markdown,
+			Title:   fmt.Sprintf("%d: %s", body.Resource.PullRequestID, body.Resource.Title),
+			Fields: []*model.SlackAttachmentField{
+				{
+					Title: "Target Branch",
+					Value: strings.Split(body.Resource.TargetRefName, "/")[2],
+					Short: true,
+				},
+				{
+					Title: "Source Branch",
+					Value: strings.Split(body.Resource.SourceRefName, "/")[2],
+					Short: true,
+				},
+				{
+					Title: "Reviewer(s)",
+					Value: reviewers,
+				},
+			},
+			Footer: body.Resource.Repository.Name,
+		}
+	case constants.PullRequestCommented:
+		reviewers := "None"
+		for i := 0; i < len(body.Resource.PullRequest.Reviewers); i++ {
+			if i == 0 {
+				reviewers = ""
+			}
+
+			if i != len(body.Resource.PullRequest.Reviewers)-1 {
+				reviewers += fmt.Sprintf("%s, ", body.Resource.PullRequest.Reviewers[i].DisplayName)
+			} else {
+				reviewers += body.Resource.PullRequest.Reviewers[i].DisplayName
+			}
+		}
+
+		attachment = &model.SlackAttachment{
+			Pretext: body.Message.Markdown,
+			Title:   fmt.Sprintf("%d: %s", body.Resource.PullRequest.PullRequestID, body.Resource.PullRequest.Title),
+			Fields: []*model.SlackAttachmentField{
+				{
+					Title: "Target Branch",
+					Value: strings.Split(body.Resource.PullRequest.TargetRefName, "/")[2],
+					Short: true,
+				},
+				{
+					Title: "Source Branch",
+					Value: strings.Split(body.Resource.PullRequest.SourceRefName, "/")[2],
+					Short: true,
+				},
+				{
+					Title: "Reviewer(s)",
+					Value: reviewers,
+				},
+				{
+					Title: "Comment",
+					Value: body.Resource.Comment.Content,
+				},
+			},
+			Footer: body.Resource.PullRequest.Repository.Name,
+		}
+	case constants.CodePushed:
+		commits := "None" // When no commits are present
+		for i := 0; i < len(body.Resource.Commits); i++ {
+			if i == 0 {
+				commits = ""
+			}
+
+			commits += fmt.Sprintf("\n[%s](%s) : **%s**", body.Resource.Commits[i].CommitID, body.Resource.Commits[i].URL, body.Resource.Commits[i].Comment)
+		}
+
+		attachment = &model.SlackAttachment{
+			Pretext:    body.Message.Markdown,
+			Title:      "Commit(s)",
+			Text:       commits,
+			Footer:     fmt.Sprintf("%s | %s", strings.Split(body.Resource.RefUpdates[0].Name, "/")[2], body.Resource.Repository.Name),
```

Comment:
here also for length, are we sure about its length?

Reply:
Yes